arangod/RocksDBEngine/RocksDBCollection.cpp

```diff
@@ -869,18 +893,29 @@ bool RocksDBCollection::readDocument(transaction::Methods* trx,
       if (ps.IsPinned()) {
         buffer->assign(ps.data(), ps.size());
       } // else value is already assigned
-      return true;
+      ret = true;;
     }
   }
-  return false;
+  _statistics._rocksdb_read_msec.count(duration<float,std::micro>(clock::now() - start).count());
+  return ret;
 }
 
 Result RocksDBCollection::insert(arangodb::transaction::Methods* trx,
                                  arangodb::velocypack::Slice const slice,
                                  arangodb::ManagedDocumentResult& resultMdr,
                                  OperationOptions& options) {
-  bool const isEdgeCollection = (TRI_COL_TYPE_EDGE == _logicalCollection.type());
 
+  if (options.isSynchronousReplicationFrom.empty()) {
+    ++_statistics._numWrite;
+  } else {
+    ++_statistics._numReplicate;
+  }
+
+  using namespace std::chrono;
+  using clock = steady_clock;
+  auto start = clock::now();
```

Comment:
If we exit the `insert()` method early (it has mutliple exits), then we increase the write counter but don't update the timers. This is inconsistent.
It would be ok if `_numWrite` / `_numReplicate` were labeled "attempted writes" / "attempted replication writes", but currently they aren't.

Reply:
the plan was to record the successful insert times in the histogram. this is detached from the counts for replication / write

Reply:
The counts are now documented to be write *operations* rather than writes and the histograms document to be *successful* operations